src/service/viewport-impl.js

```diff
@@ -290,10 +296,11 @@ export class Viewport {
 
   /**
    * Returns the size of the viewport.
+   * @param {SizeReportingFidelity} fidelity
    * @return {!{width: number, height: number}}
    */
-  getSize() {
-    if (this.size_) {
+  getSize(fidelity = SizeReportingFidelity.LOW) {
+    if (this.size_ && fidelity == SizeReportingFidelity.LOW) {
```

Comment:
I see this [code](https://github.com/ampproject/amphtml/blob/33581124eea7f43e862426b5ec765e232c6aeae9/src/service/viewport-impl.js#L911). It looks like the size is reset and immediately re-requested, which should calculate it based on the freshest data. Right?